jx/ui/FillText.jxml

```diff
@@ -2,10 +2,22 @@
 	<script>
 		<![CDATA[
 			function onAttr(attr, delta) {
-				if ( delta.text || attr.text
-					&& (delta.height || attr.height) ) {
-					delta.fontSize = delta.height || attr.height || 12;
+				if (attr.text && attr.height && attr.width) {
+					var ratio = estimate_ratio(attr.text);
+
+					if ( ratio <= (attr.width / attr.height) )
+						delta.fontSize = attr.height;
+					else
+						delta.fontSize = attr.width / ratio;
 				}
+
+				if ('textValign' in attr) {
+					delta.textValign = attr.textValign;
+				}
```

Comment:
If `attr` already has `textValign`, `delta` must have had it now or at some point in the past. So this seems redundant.